artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessage.java

```diff
@@ -1562,7 +1562,7 @@ public final Object getObjectProperty(String key) {
             return getAMQPUserID();
          case MessageUtil.CORRELATIONID_HEADER_NAME_STRING:
             if (properties != null && properties.getCorrelationId() != null) {
-               return AMQPMessageIdHelper.INSTANCE.toCorrelationIdString(properties.getCorrelationId());
+               return AMQPMessageIdHelper.INSTANCE.toCorrelationIdStringOrBytes(properties.getCorrelationId());
```

Comment:
This does change the semantics of the original `toCorrelationIdString`. Now if the ID is a `Binary` it will return a `byte[]` instead of a `String`. If we want to support compatibility between JMS clients as well as other protocols which use `byte[]` correlation IDs I don't see a way around a change like this. Currently no JMS client is compatible with another using `setJMSCorrelationIDAsBytes`.